STM32Cube/Tasks/log.c

```diff
@@ -189,10 +193,45 @@ bool logDebug(const LogDataSource_t source, const char* msg, ...)
 #endif
 }
 
+static int computeFolderSize(const char *path) {
+	DIR dir;
+	FILINFO fno;
+	int nfile = 0;
+
+	if (f_opendir(&dir, path) == FR_OK) {
+		while (f_readdir(&dir, &fno) == FR_OK && fno.fname[0]) {
+			if (!(fno.fattrib & AM_DIR))
+				++nfile;
+		}
+		f_closedir(&dir);
+	}
+
+	return nfile;
+}
+
+static void initUniqueLogFileName(const char *logsPath) {
+	int nextValidFileNumber = computeFolderSize(logsPath) - 1;
+	while (snprintf(logFileName, sizeof(logFileName), "%s/%d.txt", logsPath,
```

Comment:
could make the length of logFileName a macro and use it here instead of sizeof. idk if that matters after optimization but looks nicer.